### PR TITLE
ML-326 Integrate PBblas with platform regression suite

### PR DESCRIPTION
**Pull request by RogerDev**
Create ecl folder for regression test
Add ConfTest
Add Key for test comparison

### CODE REVIEW

File: ecl/ConfTest.ecl [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+C_S1 := PBblas.trsm(Side.Ax, Lower, FALSE, Diagonal.NotUnitTri,
+                      1.0, C_L1, E1);
+C_T1 := PBblas.trsm(Side.Ax, Upper, TRUE, Diagonal.NotUnitTri,
+                      1.0, C_L1, C_S1);
+test_11 := Tests.DiffReport.Compare_Cells('TEST1 -- Multiply, Factor, Solve', X1, C_T1);
+
+
+result  := test_11;
+
+EXPORT ConfTest := WHEN(OUTPUT(result,{testname, errors}), override);
```
Comment (johnholt) [inline ecl/ConfTest.ecl:69]: I realize that you tested this, but I don't followit.  Usually, you have WHEN(<dataset>, <action>) so I would have expected WHEN(result, OUTPUT(result, {testname, errors})); 
which will perform the output action and return the difference which is then is treated as an implicit action to output the result dataset.  This returns the action and performs the implicit action to output the value of override.

@AttilaVamos  Is the point that you want to output a dataset with just the unchanging test name and errors value?  If so, EXPORT ConfTest := OUTPUT(result, {testname, errors}); does that directly.

Comment (AttilaVamos) [reply ecl/ConfTest.ecl:69]: Yes, that is the point. The test case generates output which is changes if and only if something went wrong. In this case the errors != 0 and the result of a simple diff between the content of a specific key file and the test case output can determine the test pass or fail.
So I think the 
``` EXPORT ConfTest := OUTPUT(result, {testname, errors}); ```
would be fine.

**Comment by RogerDev** [comment]
The purpose of the WHEN is to cause the override function to get executed, which sets a low partition size so that we exercise multiple partitions (and therefore LOOP) in the test.  The OUTPUT is used to select just the desired fields from the result.

**Comment by johnholt** [comment]
That is subtle, at least to me.  For those of us that are a little slow, please add a comment to the override definition that you are using the action fired by the WHEN to evaluate _override_ and thereby  set the stored variable.  I'd like @ghalliday  to opine on whether this use case of WHEN is something that he expected and includes in the platform regression suite.  I want to make sure that we are not exploiting a side effect of the current implementation.

File: ecl/ConfTest.ecl (hunk 1 above)
Comment (ghalliday) [reply ecl/ConfTest.ecl:69]: I think the code is correct - it is a way of associating an action (or a #option/#stored) with an expression or action.

Its use in conjunction with a #option or #stored is a bit ugly, but there isn't currently another way of expressing the same thing.

**Comment by AttilaVamos** [comment]
@johnholt when do you plan to merge this PR?

**Comment by johnholt** [comment]
@richardkchapman  This PR is good to merge.

**Pull request merged**